network/src/peer_manager/mod.rs

```diff
@@ -517,7 +517,8 @@ where
                         );
                     }
                 } else {
-                    self.dial_peer(requested_peer_id, addr, response_tx).await;
+                    let request = TransportRequest::DialPeer(requested_peer_id, addr, response_tx);
+                    self.transport_reqs_tx.send(request).await.unwrap();
```

Comment:
inling code ❤️

Reply:
a lot of opportunity in here :P

I would bet that the above line never occurs, except in random conditions since the check is /the same/ as connection manager.... sigh...